web/src/app/schedules/calendar-subscribe/CalendarSubscribeCreateDialog.js

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react'
+import { PropTypes as p } from 'prop-types'
+import { useMutation } from '@apollo/react-hooks'
+import gql from 'graphql-tag'
+import FormDialog from '../../dialogs/FormDialog'
+import CalendarSubscribeForm from './CalendarSubscribeForm'
+import { getForm, FormTitle, getSubtitle } from './formHelper'
+import { fieldErrors, nonFieldErrors } from '../../util/errutil'
+
+const mutation = gql`
+  mutation($input: CreateUserCalendarSubscriptionInput!) {
+    createUserCalendarSubscription(input: $input) {
+      id
+    }
+  }
+`
+
+const MOCK_URL =
+  'www.calendarlabs.com/ical-calendar/ics/22/Chicago_Cubs_-_MLB.ics'
+
+const SUBTITLE =
+  'Create a unique iCalendar subscription URL that can be used in your preferred calendar application.'
+
+export default function CalendarSubscribeCreateDialog(props) {
+  const [isComplete, setIsComplete] = useState(false)
+  const [value, setValue] = useState({
+    name: '',
+    scheduleID: props.scheduleID || null,
+    reminderMinutes: [],
+  })
+
+  const [createSubscription, status] = useMutation(mutation, {
+    variables: {
+      input: {
+        scheduleID: value.scheduleID,
+        name: value.name,
+        reminderMinutes: value.reminderMinutes.map(r => parseInt(r.value)),
```

Comment:
I haven't come across any issues with parsing integers in base 10 before, but sounds good.

Reply:
In all honesty, we probably wouldn't likely have a problem, just a best-practice thing:

> Although discouraged by ECMAScript 3 and forbidden by ECMAScript 5, many implementations interpret a numeric string beginning with a leading 0 as octal. The following may have an octal result, or it may have a decimal result.  Always specify a radix to avoid this unreliable behavior.

From: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt

Reply:
Gotcha, I’ll keep that in mind for future parsing of the ints 😋